mlfinlab/feature_importance/cfi.py

```diff
@@ -0,0 +1,113 @@
+"""
+An Implementation of the Clustered Feature Importance algorithm described by
+Dr Marcos Lopez de Prado in 'Clustered Feature Importance (Presentation Slides)'
+(https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3517595)
+"""
+
+#Imports
+import numpy as np
+import pandas as pd
+import mlfinlab as ml
+from sklearn.metrics import log_loss
+from scipy.cluster.hierarchy import linkage, fcluster
+from information_matrix import  get_dependence_matrix , get_distance_matrix
+
+
+def cluster_features(X : pd.DataFrame , linkage_method : str , n_clusters : int = None ):
+    '''
+    Get Clusters containing features subsets
```

Comment:
Lets include way more details in these doc strings. This part of the code is almost the most important as it tells people what it does and how it is used.